tools/test/examples/examples_lib.py

```diff
@@ -0,0 +1,210 @@
+
+def print_compilation_summary(results):
+    """Prints to screen the results of compiling combinations of example programs,
+       targets and compile chains.
+
+    Args:
+    results - results of the compilation stage. See compile_repos() for
+              details of the format. 
+
+    """
+
+    print("#"*80)
+    print("# Examples compilation summary")
+    print("#"*80)
+    print("#")
+    print("# Passed example combinations")
+    print("#")
+    for key, val in results.iteritems():
+        print_list(val[1])
+            
+    print("#")
+    print("# Failed example combinations")
+    print("#")
+    for key, val in results.iteritems():
+        print_list(val[2])
+    print("#")
+    print("#"*80)
+    
+
+def target_cross_toolchain(allowed_toolchains,
+                           features=[], targets=[]):
+    """Generate pairs of target and toolchains
+
+    Args:
+    allowed_toolchains - a list of all possible toolchains
+
+    Kwargs:
+    features - the features that must be in the features array of a
+               target
+    targets - a list of available targets
+    """
+    if len(targets) == 0:
+        targets=TARGET_MAP.keys()
+        
+    for target, toolchains in get_mbed_official_release("5"):
+        for toolchain in toolchains:
+            if (toolchain in allowed_toolchains and
+                target in targets and
+                all(feature in TARGET_MAP[target].features
+                    for feature in features)):
+                yield target, toolchain
+
+
+def get_repo_list(example):
+    """ Returns a list of all the repos associated with the specific example in the json
+        config file.
+        If there are repos listed under the mbed section then these will be returned as a 
+        list. If not then the github single repo with be returned. 
+        NOTE: This does not currently deal with multiple examples underneath a github 
+        sourced exampe repo.
+
+    Args:
+    example - Example for which the repo list is requested
+    repos - The list of repos contained within that example in the json file 
+
+    """
+    repos = []
+    if len(example['mbed']) > 0:
+        for repo in example['mbed']:
+            repos.append(repo)
+    else:
+        repos.append(example['github'])
+    return repos
+
+def source_repos(config):
+    """ Clones each of the repos associated with the specific examples name from the 
+        json config file. Note if there is already a clone of the repo then it will first
+        be removed to ensure a clean, up to date cloning.
+    Args:
+    config - the json object imported from the file. 
+
+    """
+    print("\nImporting example repos....\n")
+    for example in config['examples']:
+        for repo in get_repo_list(example):
+            name = basename(repo)
+            if os.path.exists(name):
+                print("'%s' example directory already exists. Deleting..." % name)
+                subprocess.call(['rm', '-rf', name])
```

Comment:
Here you're making the assumption that we're running in a POSIX shell. I'd prefer to use the built in Python commands for this ([rmtree](https://docs.python.org/2/library/shutil.html#shutil.rmtree)) if possible to keep this cross platform. Is there a good reason to only allow this to run in a POSIX shell?